### ISSUE TITLE
Rclone sync bug: files are being erroneously erased/moved on the remote

### ISSUE THREAD

**Issue by DurvalMenezes** [ncw/rclone#1431]
> If you've just got a question or aren't sure if you've found a bug then please use the [rclone forum](https://forum.rclone.org/) instead of filing an issue.

Done already: see https://forum.rclone.org/t/rclone-sync-bug-files-are-being-erroneously-erased-moved-on-the-remote-causing-subsequent-rclone-syncs-to-copy-them-over-and-over-again-was-repeated-rclone-sync-commands-from-a-read-only-local-fs-copying-the-same-files-over-and-over-again/2252

> What is your rclone version (eg output from `rclone -V`)

Tried with both v1.36 and the latest beta (v1.36-98-g7ee3cfd7β), same behavior. 

> Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux amd64 (Devuan 1.0 Jessie, a systemd-less Debian Jessie fork).

> Which cloud storage system are you using? (eg Google Drive)

rclone-encrypted remote on top of Google Drive.

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

As per my forum topic linked above: 
`rclone --low-level-retries 100 --checkers 12 --transfers 12 -v -v --log-file ~/20170518Z111826_rclone_sync_REDACTED_egd --backup-dir egd:REDACTED_-_backup-dir_-_20170518Z111826 sync /REDACTED egd:REDACTED
`
> A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

Most relevant log lines are: 
```
2017/05/18 12:36:18 INFO  : REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED.info: Copied (new)
2017/05/19 10:12:27 INFO  : REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED.info: Moved (server side)
2017/05/19 10:12:27 INFO  : REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED/REDACTED.info: Moved into backup dir
```

Please also see my forum topic above for a longer explanation, specially this post where I figured out the reason for the repeated copyings: https://forum.rclone.org/t/rclone-sync-bug-files-are-being-erroneously-erased-moved-on-the-remote-causing-subsequent-rclone-syncs-to-copy-them-over-and-over-again-was-repeated-rclone-sync-commands-from-a-read-only-local-fs-copying-the-same-files-over-and-over-again/2252/6?u=durval (specially the paragraph starting with "EDIT").

**Comment by ncw** [comment]
Can you try running [rclone dedupe](https://rclone.org/commands/rclone_dedupe/) on the underlying google drive (not the egd: but the one its `remote` points to), eg something like

    rclone dedupe --dry-run gd:whatever_you_are_using

Drive is a bit prone to duplicating files that you upload which `dedupe` can fix.  Those would cause the symptoms above I think.  Unlike a normal filing system, drive can have two files with the **same** name!

That might not be the problem, but if you can rule that out first that would be excellent.

**Comment by DurvalMenezes** [comment]
Hi @ncw, 

Thanks for your response. Oh my, even after running GDrive intensively for over 6 months I wasn't aware it could get lost and produce two files with the same name! this sounds completely absurd to me. Well, it's "welcome to the real world" for me, I guess... thanks for that blue pill! ;-)

Two questions:
1) The unencrypted remote my "egd" is built on is indeed called "gd", but what do I replace the "whatever_you_are_using" part with? I would guess I should replace it with the encrypted name for the directory  I'm verifying (and getting these issues on), ie, the "REDACTED"  in my "egd:REDACTED" `rclone sync` command line. Is that correct? Or should I just leave it empty (and let `rclone dedupe` work its away through the entire gd contents)? 

2) Out of curiosity, why TF could have caused GDrive to create those duplicates? Collisions on the rclone filename encryption? Some kind of internal bug/feature, perhaps interacting weirdly with rclone? Inquiring minds want to know ;-)

Thanks, 
--   Durval.

**Comment by RomeSilvanus** [comment]
I do have a similar problem. For me it is synching files down to my server, but then erasing them again at the end of the sync.

The dedupe command finds no duplicates. -q doesn't print any errors.
I uploaded the files with Google Drive web originally.

I really don't know where else to look for an error.

**Comment by DurvalMenezes** [comment]
> I do have a similar problem. For me it is synching files down to my server, but then erasing them again at the end of the sync.

This is *exactly* what is happening with me, see above.

> The dedupe command finds no duplicates. -q doesn't print any errors.

Can you please share exactly what your `rclone dedup` command was?

> I uploaded the files with Google Drive web originally.

My data wa s uploaded completely with `rclone copy` to an encrypted remote: initially from local to ACD, then from ACD to GDrive, then from local to GDrive again (as the initial copy to ACD was interrupted in the middle due to my migration to GDrive). Only after the copy to GDrive finished with no errors I started using sync.

> I really don't know where else to look for an error.

Neither do I, but do not despair: @ncw will fix it in no time! :-) 

Cheers,
-- Durval.

**Comment by RomeSilvanus** [comment]
My files are not encrypted. These files were uploaded from my Mac to GDrive through their web interface.

Dedupe command:
$ rclone dedupe -v Google:Pony/Episodes
2017/05/24 14:22:56 INFO  : Google drive root 'Pony/Episodes': Modify window is 1ms
2017/05/24 14:22:56 INFO  : Google drive root 'Pony/Episodes': Looking for duplicates using interactive mode.
$

**Comment by alternativesurfer** [comment]
Can confirm this is also affecting me.
Recently switched to Google Drive from ACD.
Successfully synced a few files last Friday, tested, played back fine.

Looks like from Sunday on this bug began deleting files after upload completed.

Ran the dedupe command recommended above, did not find any duplicates nor report errors.
Ran: rclone dedupe --dry-run google:mynamehere

**Comment by ncw** [comment]
@DurvalMenezes wrote
> The unencrypted remote my "egd" is built on is indeed called "gd", but what do I replace the "whatever_you_are_using" part with? I would guess I should replace it with the encrypted name for the directory I'm verifying (and getting these issues on), ie, the "REDACTED" in my "egd:REDACTED" rclone sync command line. Is that correct? Or should I just leave it empty (and let rclone dedupe work its away through the entire gd contents)?

You can use the remote that you put in the crypt config to check the whole thing, or point at the subdirectory if you can work out what it is called

> Out of curiosity, WTF could have caused GDrive to create those duplicates? Collisions on the rclone filename encryption? Some kind of internal bug/feature, perhaps interacting weirdly with rclone? Inquiring minds want to know ;-)

I wish I knews - it is one of rclone's oldest bugs #28.  I did a bit of investigation with the help of another user and Google - I think it may be caused by files not appearing in directory listings occasionally, but the jury is out!

**Comment by ncw** [comment]
@alternativesurfer @RomeSilvanus 
I need to see logs with `-vv`.  If you can grep out logs all logs which mention one affected file name and paste them here that would be very useful.  rclone will say why it is uploading and deleting stuff somewhere in the log hopefully!  Also which parameters you are using with rclone and whether you are using crypt

**Comment by alternativesurfer** [comment]
@ncw 
Logs: https://pastebin.com/qc2Lz24N

Used 1 test file.

Rclone params:
/usr/sbin/rclone mount --dir-cache-time=5m --read-only --no-modtime  --allow-non-empty --allow-other google:plex /home/plex/.acd &

**Comment by RomeSilvanus** [comment]
@ncw 

Rclone cmd:
`rclone copy -vv Google:Pony/Music/Fan-made/Equestrian\ Beats\ \(EQBeats.org\)\ full\ rip/Artists/IvoryKeysADSR/ www/iwatchitfortheplot/Pony/Music/Fan-made/Equestrian\ Beats\ \(EQBeats.org\)\ full\ rip/Artists/IvoryKeysADSR/`
(This is only one of the folders that gave an error)

I hope this is what you were asking, unless you mean a log from somewhere else:
Here's a cleaned up version: https://pastebin.com/eMvAHJba
Here's the full original output: https://pastebin.com/zHHGErcd

**Comment by RomeSilvanus** [comment]
@ncw 

Here is the full log with -vv : https://pastebin.com/GZthkiE5
It doesn't tell me _why_ it is deleting these files, unless I'm missing something

rclone dedupe didn't yield any results on these files. And as I said even when I successfully copy them with rclone copy it will still delete them when using rclone snyc.

**Comment by ncw** [comment]
@alternativesurfer in your log you are using `rclone move` so deleting the file (locally) after it has been copied (remotely) is exactly what I'd expect to see.

@RomeSilvanus your problem is caused by OS X and denormalised UTF-8.  When you uploaded the files to drive using the Gdrive app on Mac it sent them up with denormalised UTF-8 ([tech details](https://stackoverflow.com/questions/9757843/unicode-encoding-for-filesystem-in-mac-os-x-not-correct-in-python)).

Here are the log messages

```
2017/05/26 20:26:24 INFO  : My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4: Copied (new)
2017/05/26 20:26:24 INFO  : My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4: Deleted
```

Copying those names into python you can see they are different though they appear the same in the browser.

```
>>> a=u"My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4"
>>> b=u"My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4"
>>> a == b
False
>>> a
u'My Little Pony FiM - 6.25 - Die Ru\u0308ckkehr der Wechselponys - Teil 1.mp4'
>>> b
u'My Little Pony FiM - 6.25 - Die R\xfcckkehr der Wechselponys - Teil 1.mp4'
>>> 
```

rclone normalises the file names so you should find it wants to upload each files once only.

However you can use the `--track-renames` flag to avoid rclone uploading the data again - one sync should be sufficient to rename and normalise everything.

@DurvalMenezes The latter could possibly be related to your problem.  Either way can you make a small demo of the problem that I can reproduce locally?  I'm having trouble working out what is going on!

**Comment by ncw** [comment]
@alternativesurfer in your log you are using `rclone move` so deleting the file (locally) after it has been copied (remotely) is exactly what I'd expect to see.

@RomeSilvanus your problem is caused by OS X and denormalised UTF-8.  When you uploaded the files to drive using the Gdrive app on Mac it sent them up with denormalised UTF-8 ([tech details](https://stackoverflow.com/questions/9757843/unicode-encoding-for-filesystem-in-mac-os-x-not-correct-in-python)).

Here are the log messages

```
2017/05/26 20:26:24 INFO  : My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4: Copied (new)
2017/05/26 20:26:24 INFO  : My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4: Deleted
```

Copying those names into python you can see they are different though they appear the same in the browser.

```
>>> a=u"My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4"
>>> b=u"My Little Pony FiM - 6.25 - Die Rückkehr der Wechselponys - Teil 1.mp4"
>>> a == b
False
>>> a
u'My Little Pony FiM - 6.25 - Die Ru\u0308ckkehr der Wechselponys - Teil 1.mp4'
>>> b
u'My Little Pony FiM - 6.25 - Die R\xfcckkehr der Wechselponys - Teil 1.mp4'
>>> 
```

rclone normalises the file names so you should find it wants to upload each files once only.

However you can use the `--track-renames` flag to avoid rclone uploading the data again - one sync should be sufficient to rename and normalise everything.

@DurvalMenezes The latter could possibly be related to your problem.  Either way can you make a small demo of the problem that I can reproduce locally?  I'm having trouble working out what is going on!

**Comment by alternativesurfer** [comment]
@ncw

The delete you are seeing is executing on the remote file. 
I can redo this with "copy" to prove it if necessary. 
I have used Move successfully​ for months now, only recently did it begin deleting the remote copies. 

I could sync a dozen files and at the end rsync would "delete" the remote copy, leaving me with no files in Drive.

**Comment by ncw** [comment]
@alternativesurfer I'd like to see a log with that if possible, or best of all make a small scenario I can reproduce it with.  I haven't had any luck reproducing the problem yet.

**Comment by alternativesurfer** [comment]
Okay, so I performed it with a copy and it didn't log the Delete at the end so you were correct.
I don't know why the behavior changed but I'll change my workflow and use Copy moving forward.

New weird thing (and maybe I'll need to open a new issue rather than working on this):
My mount does not appear to be working.
I can upload fine now (proved by viewing the encrypted files in Drive).
However the mount must be failing because when I navigate to that folder in my server there is nothing present...
Cest La Vie, haha

**Comment by ncw** [comment]
@alternativesurfer have you got a stale mount open?

**Comment by ncw** [comment]
I just had an idea about this issue.

I wonder if you have duplicate directories on Google drive. Rclone dedupe only detects duplicate files, however duplicate directories could have a similar effect.

    rclone lsd --max-depth 999 remote:

Will list all the directories. A bit of cut, sort, uniq will find any duplicates (not at pc right now so can't provide a recipe)

**Comment by RomeSilvanus** [comment]
@ncw Thanks!

I actually did `copy` it down to my server and then `sync`ed it back up to GDrive and that fixed everything. (Normally I only sync from GDrive down)

Though maybe Rclone should tell you more why it failed instead of leaving you up to wonder what exactly went wrong. I can see it happening more often if OSX actually works like this.

**Comment by ncw** [comment]
@RomeSilvanus glad you fixed it. I guess I could either put a warning in about unnormalised utf-8, or normalise it.  I made an issue here #1457

@DurvalMenezes  can you run this to see if you have any duplicated directories

    rclone lsd remote:path | cut -b 44- | sort | uniq -c | sort -n | grep -v ' *1 '

**Comment by alternativesurfer** [comment]
@ncw 
> @alternativesurfer have you got a stale mount open?

I honestly am not even sure how to confirm that.
I have restarted the machine a few times though, which I would assume unmounts / remounts it?

**Comment by johnl** [comment]
I'm having this problem too, syncing a local directory tree to an OpenStack Swift server. There is a "directory" on the swift server that rclone wants to erroneously delete the files from. It first finds the files match and are ok, then tries to delete them:

```
2017/06/06 18:02:47 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
2017/06/06 18:02:47 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping
2017/06/06 18:02:48 NOTICE: 110428 - Mums Birthday Party/DSC_5426.JPG: Not deleting as --dry-run
2017/06/06 18:02:48 INFO  : 110428 - Mums Birthday Party/DSC_5426.JPG: Deleted
```

I ran the suggested lsd command and that directory does appear to have a duplicate but it's not actually a duplicate. The dir has a subdirectory (with a few copies of the files from it's parent) so it shows up twice with the lsd list.

I've confirmed with curl that there really is no duplicate (no file with the same name as the directory for example).

The container has hundreds of "directories" and many tens of thousands of objects, and many other directories with subdirectories.

I've been unable to replicate it elsewhere - even just copying this exact directory over elsewhere.

Might it be related to a large number of files? Swift's object list limit is 10000, so rclone is presumably paging through the object list - perhaps a bug there?

so far I've only run this with dry-run and aren't in a rush, so I can keep running tests if you like. But I think I've run out of things to try myself here.

**Comment by ncw** [comment]
@johnl sorry for the delay in replying - what is your rclone command line?  I'm having trouble working out the sequence of events which could cause those log messages unless you are using `rclone move`?

> Might it be related to a large number of files? Swift's object list limit is 10000, so rclone is presumably paging through the object list - perhaps a bug there?

rclone lists in chunks of 1000 (that was 256 until recently)

Can you try again with the latest beta to see if the problem is still there?

**Comment by johnl** [comment]
I'm using `rclone sync`. I'm using `--include` here just to focus on the problematic directory (as there are a lot of others) but it happens if I run it without and just grep the output for that directory:

    rclone -vvv -n --delete-during sync --include 110428\ -\ Mums\ Birthday\ Party/\* . orbit:photos/photos/john/

If I change to --delete-after the order of the logs is different (as you'd expect):

    2017/06/28 23:14:10 NOTICE: 110428 - Mums Birthday Party/DSC_5426.JPG: Not deleting as --dry-run
    2017/06/28 23:14:10 INFO  : 110428 - Mums Birthday Party/DSC_5426.JPG: Deleted
    2017/06/28 23:14:11 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
    2017/06/28 23:14:11 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping

Upgraded to the beta release dated 27th June (v1.36-222-gf3c7e1a9β) and it looks fixed!

    2017/06/28 23:19:36 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
    2017/06/28 23:19:36 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping

**Comment by ncw** [comment]
@johnl Just trying to work out what happened - which version of rclone were you using originally?

**Comment by johnl** [comment]
I was originally using official amd64 build of rclone v1.36, dated March 18. Running on Debian Jessie.

I've just tested it with every build available on beta.rclone.org in order and it's first fixed in release v1.36-185-g64662bef

```
beta.rclone.org/v1.36-174-g1e88f070/rclone-v1.36-174-g1e88f070β-linux-amd64.zip
2017/06/29 11:40:21 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
2017/06/29 11:40:21 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping
2017/06/29 11:40:21 NOTICE: 110428 - Mums Birthday Party/DSC_5426.JPG: Not deleting as --dry-run

beta.rclone.org/v1.36-185-g64662bef/rclone-v1.36-185-g64662befβ-linux-amd64.zip
2017/06/29 11:40:23 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
2017/06/29 11:40:23 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping

beta.rclone.org/v1.36-186-g9876ba53/rclone-v1.36-186-g9876ba53β-linux-amd64.zip
2017/06/29 11:40:24 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Size and modification time the same (differ by 0s, within tolerance 1ns)
2017/06/29 11:40:24 DEBUG : 110428 - Mums Birthday Party/DSC_5426.JPG: Unchanged skipping
```

Perhaps worth noting also that the output changed in beta.rclone.org/v1.36-22-gbc25190 and it stopped logging the INFO "Deleted" line. Arguably better - but wanted to mention in case relevant.

**Comment by DurvalMenezes** [comment]
Hello everyone, (and @ncw specially),

Finally got the leisure to get back on this. 

So, following Nick's suggestion, yesterday I ran this: 

`rclone lsd --max-depth 999 egd: | cut -b 44- | sort | uniq -c | sort -n | grep -v ' *1 '`

("egd" is my encrypted Google Drive remote, currently with ~16.5TB and a few million files and probably -- haven't counted -- a few hundred thousand directories). 

The result, after a day and change of running, is: 
```
      2 REDACTED1/REDACTED2/REDACTED3/REDACTED4/REDACTED5/REDACTED6/REDACTED7/REDACTED8
      2 REDACTED9/REDACTED10/REDACTED11/REDACTED12/REDACTED13/REDACTED14/REDACTED15/REDACTED16/REDACTED17/REDACTED18/REDACTED19/REDACTED20
      2 REDACTED9/REDACTED10/REDACTED11/REDATED21/REDACTED22/REDACTED23/REDACTED24/REDACTED25
      2 REDACTED9/REDACTED10/REDACTED11/REDACTED26/REDACTED27/REDACTED28/REDACTED29/REDACTED30/REDACTED31
      2 REDACTED32/REDACTED33/REDACTED34/REDACTED35/REDACTED36/REDACTED37/REDACTED38
      2 REDACTED32/REDACTED33/REDACTED34/REDACTED35/REDACTED39/REDACTED40/REDACTED41/REDACTED42/REDACTED43/REDACTED44
      2 REDACTED32/REDACTED33/REDACTED34/REDACTED45/REDACTED46/REDACTED47/REDACTED48
      2 REDACTED32/REDACTED33/REDACTED34/REDACTED45/REDACTED49/REDACTED50/REDACTED51/REDACTED52/REDACTED53
      2 REDACTED32/REDACTED33/REDACTED34/REDACTED54/REDACTED55/REDACTED56/REDACTED57/REDACTED58/REDACTED59/REDACTED60/REDACTED61/REDACTED62/REDACTED63/REDACTED64/REDACTED65/REDACTED66/REDACTED67/REDACTED68/REDACTED69/REDACTED70
```

So, 9 duplicated directories, each with exactly 2 occurrences,  spread among 3 of the top-level directories in that remote (the fact that 5 of them are located on the same top-level dir, namely REDACTED32 above, is expected as this is by far the heaviest -- also in terms of bytes used, but specially in terms of number of files and depth and breadth of its directory tree). 

I cross-checked the above list with the list of files that I reported as being repeatedly deleted at the end of each `rclone sync` (and then copied back at the next `rclone sync`, just to be deleted again at the end of it), and the results are:

-> The first 4 directories listed above have *zero* of those deleted/recopied files under them, but then they were outside of the dirtree being updated by the `rclone sync` in question;
-> The others are as follows:
directory Nr. 5: 49 files
directory Nr. 6: 4 files
directory Nr. 7: 2 files
directory Nr. 8: 2 file
directory Nr, 9: 11 file
*Total: 68 files*

As the total number of files that was recopied and then redeleted in that run was exactly 68, I think we can chalk all of them (and therefore, hopefully, my entire trouble) to these duplicated directories, as @ncw suspected.

I have two more questions:

1) How to solve that problem? Delete those duplicated directories and then repeat the sync? Would a simple "rclone purge" on each one of them suffice? 

2) I understand that Google Drive produces these duplicated directories "semi expontaneously" (ie, independent of rclone), is that correct? In that case, would running a `rclone lsd --max-depth 999 egd: | cut -b 44- | sort | uniq -c | sort -n | grep -v ' *1 '` periodically (say, once a week) be enough to detect them so I could bring the above purgehammer to bear? 

Thanks, 
-- Durval.

**Comment by ncw** [comment]
> How to solve that problem? Delete those duplicated directories and then repeat the sync? Would a simple "rclone purge" on each one of them (on the remote side, of course) suffice?

If you try [the latest beta](https://beta.rclone.org) you'll find that `rclone dedupe` now merges identical directories.  Try it with `--dry-run` first as it is new code!

> I understand that Google Drive produces these duplicated directories "semi expontaneously"

Yes it seems to be random.  I suspect some sort of not up to date caching in drive.

Running `rclone dedupe --dry-run` will detect them also quite well.

**Comment by DurvalMenezes** [comment]
Hello @ncw, 

> If you try the latest beta you'll find that rclone dedupe now merges identical directories. Try it with --dry-run first as it is new code!

But would just a "rclone purge" on each of those directories do the trick? A dedupe will certainly take more than a day to run... :-/

Cheers, 
-- Durval.

**Comment by ncw** [comment]
> But would just a "rclone purge" on each of those directories do the trick? A dedupe will certainly take more than a day to run... :-/

An rclone purge will delete one of the directories... Which one I'm not so sure!

**Comment by DurvalMenezes** [comment]
Hello @ncw , 

> An rclone purge will delete one of the directories... Which one I'm not so sure!
None of them are very critical, and I have local copies of all of them just in case... and they are reasonably small byte-wise, so even if "rclone purge" deleted both, it would not be a problem...

I will "rclone purge" them and then I will run the "rclone lsd (...)" one-liner above to confirm there are no more duplicates, and then I will run an "rclone sync" on the whole she-bang. 

Expect a report in one week or so (the "rclone sync" of that directory tree really takes a *long* time to run).

Cheers, 
-- Durval.

**Comment by DurvalMenezes** [comment]
Just keeping everyone updated, the individual "rclone purge" commands went through allright, and an "rclone lsd" which I ran immediately after confirmed that there were no more duplicates.

Right now I'm doing the "rclone sync" on that big dirtree, but having issues with it never finishing and just checking the same files over and over again... more details here: https://forum.rclone.org/t/rclone-sync-never-finishes-keeps-going-through-the-same-files-over-and-over-again/3570/1

**Comment by DurvalMenezes** [comment]
Hello, 

The new "rclone sync"  finished yesterday, with many files Checked and Transferred, as expected, and no Errors;  also, the backup-dir quite filled up (123269 items and 219.014 GB), also as expected.

Then I re-ran the exactly same "rsync clone" command (changing just the backup-dir and the log-file names) and it finished today, with many files Checked, and *no* Transferred and *no* Errors; so I think the original "files are being erroneusly erased/moved" problem that brought me to opening this issue has been fixed (to wit, by scanning for and then purging the duplicated directories that Google Drive created by itself, see above), and therefore I'm closing this.  

Thanks again @ncw for your great support, and for the great piece of software that is rclone.

**Comment by ncw** [comment]
Apologies for the delay in responding, I've been on vacation.

Thank you for too your patience and lots of testing - much appreciated :-)

**Comment by ncw** [comment]
@DurvalMenezes  I'll just note that while `rclone dedupe` or the manual equivalent fixed the problem with the duplicates, there was a problem with rclone's handling of duplicate files, now fixed in 00fe6d95daadd96b272266faa5441f4f3390f331 which was causing spurious copies and deletes.
